src/app/_services/metadata.service.ts

```diff
@@ -0,0 +1,261 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { HttpHeaders, HttpClient } from '@angular/common/http';
+
+import { AuthService, Asset } from 'app/shared';
+import { map, mergeMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MetadataService {
+
+  constructor(
+      private _auth: AuthService,
+      private _http: HttpClient
+  ) { }
+
+
+    public buildAsset(assetId: string, { groupId = '', legacyFlag = true, openlib = false, encrypted = false }={}): Observable<Asset> {
```

Comment:
Do we want to set legacyFlag to true here ?

Reply:
Good catch...